packages/hardhat-core/test/internal/hardhat-network/helpers/useProvider.ts

```diff
@@ -115,6 +122,79 @@ export function useProvider({
         this.server.getProvider()
       );
     }
+
+    if (rethnetBinary !== undefined) {
+      this.rethnetProcess = spawn(normalize(rethnetBinary), ["node", "-vv"]);
+
+      let stdout = "";
+      this.rethnetProcess.stdout.on("data", (data: any) => {
+        stdout += data.toString();
+        console.log(`rethnet subprocess ${this.rethnetProcess.pid}: ${data}`);
+      });
+
+      let stderr = "";
+      this.rethnetProcess.stderr.on("data", (data: any) => {
+        stderr += data.toString();
+        console.log(`rethnet subprocess ${this.rethnetProcess.pid}: ${data}`);
+      });
+
+      function outputForError() {
+        return `stdout:\n${stdout}\nstderr:\n${stderr}`;
+      }
+
+      const wait = new Promise((resolve) => setTimeout(resolve, 2000));
+
+      const exit = new Promise<void>((resolve, reject) => {
+        this.rethnetProcess.on("exit", (code: number, signal: string) => {
+          if (signal === "SIGKILL") {
+            console.log("kill");
+            resolve();
+          } else {
+            reject(
+              new Error(
+                `rethnet process closed unexpectedly. return code: ${code}. signal: ${signal}. ${outputForError()}`
+              )
+            );
+          }
+        });
+      });
+
+      const error = new Promise((_resolve, reject) => {
+        this.rethnetProcess.on("error", (err: Error) => {
+          if (err.message.includes("ENOENT")) {
+            reject(
+              new Error(
+                `Rethnet executable not found at ${resolvePath(rethnetBinary)}`
+              )
+            );
+          } else {
+            reject(
+              new Error(`Rethnet subprocess error: ${err}. ${outputForError()}`)
+            );
+          }
+        });
+      });
+
+      // sleep a moment to let the server initialize
+      await new Promise((resolve) => setTimeout(resolve, 250));
```

Comment:
Same comment about the sleep helper.

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4095/commits/368202e32b4226e46905d22cb08899d4f2cbab97